src/lib.rs

```diff
@@ -247,16 +247,18 @@ impl<T: Debug + PrimInt + One + Zero> Vob<T> {
     /// assert!(v.capacity() >= 1);
     /// ```
     pub fn reserve(&mut self, additional: usize) {
-        // Repeatedly calling this will over-reserve, since we only record how many bits we've
-        // used, not how many bits of capacity the user thinks we have. For example, repeatedly
-        // asking for storage for one extra bit will repeatedly add a whole extra machine word.
-        // Fixing this would require an extra machine word in the Vob struct, which hardly seems
-        // worth it for an unlikely, and probably silly, use case.
-        let cap = self.vec.capacity()
-                          .checked_mul(bits_per_block::<T>())
-                          .and_then(|x| x.checked_add(additional))
-                          .expect("Overflow detected");
-        self.vec.reserve(blocks_required::<T>(cap));
+        let current_free_space: usize = self.capacity() - self.len();
```

Comment:
Should we call this something shorter like `unused_bits`? Or even just `ub` as in one of the other functions.

Reply:
I've done `unused_bits`, as `ub` turned out to mean `used_bits` in most places I checked. It's sometimes a bit confusing as doing all this bit-mangling gives me headaches.